File: AtlasWeb/Core/Modules/Tibbi_Surec/Hemodiyaliz_Modulu/InstructionListFormViewModel.ts
//$98BDA226
import { BaseViewModel } from "../../../wwwroot/app/NebulaClient/Model/BaseViewModel";
import { HemodialysisInstruction } from "NebulaClient/Model/AtlasClientModel";
import { MultipleDataComponent_SummaryInfo } from "../Tibbi_Surec_Evrensel_Modulu/BaseMultipleDataEntryFormViewModel";
import { Type } from "../../../wwwroot/app/NebulaClient/ClassTransformer";

export class InstructionListFormViewModel extends BaseViewModel {
    @Type(() => HemodialysisInstruction)
    public _HemodialysisInstruction: HemodialysisInstruction = new HemodialysisInstruction();

    @Type(() => MultipleDataComponent_SummaryInfo)
    public HemodialysisInstructionInfoList: Array<MultipleDataComponent_SummaryInfo> = new Array<MultipleDataComponent_SummaryInfo>();
}

File: AtlasWeb/Core/Modules/Tibbi_Surec/Hemodiyaliz_Modulu/InstructionListFormViewModel.cs
//$6BB38442
using System;
using System.Linq;
using Core.Models;
using Infrastructure.Filters;
using Infrastructure.Models;
using TTInstanceManagement;
using TTObjectClasses;
using TTDefinitionManagement;
using TTUtils;
using System.Collections.Generic;

namespace Core.Controllers
{
    public partial class HemodialysisInstructionServiceController
    {
        partial void PreScript_InstructionListForm(InstructionListFormViewModel viewModel, HemodialysisInstruction hemodialysisInstruction, TTObjectContext objectContext)
        {
            Guid? activeEpisodeActionObjectID = viewModel.GetSelectedEpisodeActionID();
            if (activeEpisodeActionObjectID.HasValue)
            {
                EpisodeAction episodeAction = objectContext.GetObject<EpisodeAction>(activeEpisodeActionObjectID.Value);

                var baseMultipleDataEntryServiceController = new BaseMultipleDataEntryServiceController();
                // MultipleDataComponent için Eklendiði formun presine eklenir
                int summaryLimitCount = 5;
                viewModel.HemodialysisInstructionInfoList = baseMultipleDataEntryServiceController.GetBaseMultipleDataEntrySummaryInfoList(0, summaryLimitCount, episodeAction.BaseMultipleDataEntries.ToList<BaseMultipleDataEntry>());

                hemodialysisInstruction.EpisodeAction = episodeAction;
            }
        }
    }
}

namespace Core.Models
{
    public partial class InstructionListFormViewModel : BaseViewModel
    {
        public List<MultipleDataComponent_SummaryInfo> HemodialysisInstructionInfoList  // MultipleDataComponent için
        {
            get;
            set;
        }
    }
}
